### PR TITLE
New Rule Detect Linux Cgroup Container Escape Vulnerability (CVE-2022-0492)

### PR DESCRIPTION
**Pull request by darryk10**
This PR add a new rule to detect container escapes techniques (also used CVE-2022-0492 Linux Cgroup Container Escape Vulnerability) using new field available thread.cap_effective to check capabilities in the container which is part of the requirement for the exploitation. 
Reference: 
https://unit42.paloaltonetworks.com/cve-2022-0492-cgroups/
https://blog.trailofbits.com/2019/07/19/understanding-docker-container-escapes/

Signed-off-by: darryk10 <stefano.chierici@sysdig.com>
Co-authored-by: Lorenzo Susini <susinilorenzo1@gmail.com>

**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

> /kind bug

> /kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

> /kind feature

> If contributing rules or changes to rules, please make sure to also uncomment one of the following line:

> /kind rule-update

/kind rule-create

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area build

> /area engine

/area rules

> /area tests

> /area proposals

**What this PR does / why we need it**:
This PR add a new rule to detect container escapes techniques (also used CVE-2022-0492 Linux Cgroup Container Escape Vulnerability)
**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
rule(Detect release_agent File Container Escapes): new rule created to detect an attempt to exploit a container escape using release_agent file
```

### CODE REVIEW

File: rules/falco_rules.yaml [hunk 1]
```diff
@@ -3142,6 +3142,17 @@
   priority: CRITICAL
   tags: [process, mitre_privilege_escalation]
 
+# This rule helps detect CVE-2022-0492:
+# A privilege escalation container escaping in cgroup
+- rule: Linux Cgroup Container Escape Vulnerability (CVE-2022-0492)
+  desc: "Detect an attempt to exploit a container escape vulnerability in the Linux Kernel (CVE-2022-0492). By running a container with certains capabilities, a privileged user can modify release_agent file and escape from the container"
+  condition:
+    open_write and fd.name endswith release_agent and (user.uid=0 or thread.cap_permitted contains CAP_DAC_OVERRIDE) and excessively_capable_container
```
Comment (Kaizhe) [inline rules/falco_rules.yaml:3150]: please add the `container` condition

Comment (Kaizhe) [inline rules/falco_rules.yaml:3150]: is this `excessively_capable_container` macro defined somewhere?

**Comment by Kaizhe** [comment]
The prerequisite make it hard to exploit:
```
As the root user, or without the [no_new_privs](https://kubernetes.io/docs/concepts/policy/pod-security-policy/#privilege-escalation) flag; and
Without AppArmor or [SELinux](https://www.redhat.com/en/topics/linux/what-is-selinux#:~:text=Security%2DEnhanced%20Linux%20(SELinux),Linux%20Security%20Modules%20(LSM).); and
Without [Seccomp](https://docs.docker.com/engine/security/seccomp/); and
On a host that enables unprivileged user namespaces; and
In a root v1 cgroup.
```

File: rules/falco_rules.yaml (hunk 1 above)
Comment (darryk10) [reply rules/falco_rules.yaml:3150]: the macro is introduced in this PR https://github.com/falcosecurity/falco/pull/1963/files done by @loresuso

Comment (Kaizhe) [reply rules/falco_rules.yaml:3150]: Then what non-default capabilities are needed for the exploit?

**Comment by darryk10** [comment]
Thanks @Kaizhe for the review :) 
I think it's worth noting this CVE (along with others) can be exploited in privileged container as well which increase the chance to get exploited. 
In addition  there are different container escaping techniques that use relase_agent file to reach container escaping. Some of those also found in malwares reported in the following link. The rule created checking the capabilities will trigger on those use cases as well. 
https://blog.aquasec.com/threat-alert-container-escape

**Comment by darryk10** [comment]
@Kaizhe Looking at other techniques where release_agent file is involved. I figured out the same rule is effective for all the techniques which involve release_agent file (included the CVE-2022-0492). So I redesigned the rule to be generic on the release_agent file. Hope it's better now :) 
References: 
https://blog.trailofbits.com/2019/07/19/understanding-docker-container-escapes/
https://unit42.paloaltonetworks.com/cve-2022-0492-cgroups/

Thanks again

**Review by Kaizhe — APPROVED** [review]

**Comment by leogr** [comment]
Closing and reopening to trigger the CI
/close

**Review by leogr — COMMENTED** [review]
I think that tests are failing because the driver version used by this PR is still `caa0e4d0044fdaaebab086592a97f0c7f32aeaa9`. IIRC, `thread.cap_effective` was introduced later.

If so, rebasing this PR on top of the current `master` branch should solve the issue.
@darryk10 could you rebase this PR, please?

**Review by leogr — APPROVED** [review]
Great, after the rebase, it works :+1: 


/approve

**Review by jasondellaluce — APPROVED** [review]
/approve

**Pull request merged**